### PR TITLE
qualys_vmdr: add support for configuring which cloud metadata to use

### LINKED ISSUE #10277: [qualys_vmdr.asset_host_detection] Align `cloud.*` fields with Qualys data and not elastic agent metadata
When ingesting vulnerability data, `cloud.*` fields are a mix of the Elastic agent metadata and Qualys vulnerability data. I didn't find a way to disable metadata coming from Elastic agent (`publisher_pipeline.disable_host` seems to apply to field `host` only). 

A decision must be made to either have no cloud fields set, either have only the data coming from Qualys, either from the elastic agent metadata.

I have a strong preference to remove any properties set by the elastic agent and only have data from Qualys. If we want to build dashboard with data coming from different scanning solutions (Qualys, Tenable, Snyk, ...), we'll heavily rely on ECS fields. 

We might have a rule that if the field `tags` contains `forwarded`, we use only the data from Qualys. 

Here is an example of what I'm getting
```json
"cloud": {
  "availability_zone": "us-central1-a", # Elastic agent metadata
  "instance": {
    "name": "gke-xxx", # Elastic agent metadata
    "id": "12346789" # Qualys vulnerability data
  },
  "provider": "GCP", # Qualys vulnerability data
  "service": {
    "name": "Compute Engine" # Qualys vulnerability data
  },
  "project": {
    "id": "my-project" # Elastic agent metadata
  },
  "region": "us-central1", # Elastic agent metadata
  "account": {
    "id": "my-project" # Elastic agent metadata
  }
}
```

### PR DESCRIPTION
**Pull request by efd6**
## Proposed commit message

This adds a UI configurator that allows the user to select which of the two sources (or both) to use to populate the cloud fields. It also tags the document with an indicator of which sources were available.

---

It's quite extravagant, so I'm not sure of the merits of merging this, but is was what was asked for as the ideal. So here it is.

## Checklist

- [ ] I have reviewed [tips for building integrations](https://github.com/elastic/integrations/blob/main/docs/tips_for_building_integrations.md) and this pull request is aligned with them.
- [ ] I have verified that all data streams collect metrics or logs.
- [ ] I have added an entry to my package's `changelog.yml` file.
- [ ] I have verified that Kibana version constraints are current according to [guidelines](https://github.com/elastic/elastic-package/blob/master/docs/howto/stack_version_support.md#when-to-update-the-condition).

## Author's Checklist

- [ ] 

## Related issues

- Closes #10277

## Screenshots

qualys_vmdr-cloud-field-selector

### CODE REVIEW

File: packages/qualys_vmdr/data_stream/asset_host_detection/manifest.yml [hunk 1]
```diff
@@ -48,6 +48,23 @@ streams:
...
+        title: Cloud Metadata Source
+        description: What source to use to populate `cloud.*` fields.
+        type: select
+        multi: false
+        required: true
+        show_user: false
+        options:
+          - text: Elastic Only
+            value: elastic_cloud_data
+          - text: Provider Provider
```
Comment (kcreddy) [inline packages/qualys_vmdr/data_stream/asset_host_detection/manifest.yml:61]: Should this text be just `Provider` instead of `Provider Provider`

**Review by kcreddy — APPROVED** [review]
Minor clarification

File: packages/qualys_vmdr/data_stream/asset_host_detection/manifest.yml (hunk 1 above)
Comment (efd6) [reply packages/qualys_vmdr/data_stream/asset_host_detection/manifest.yml]: Thanks. It should be "Provider Only".

**Comment by efd6** [comment]
Waiting for @clement-fouque.

**Comment by andrewkroh** [comment]
fyi: I added a screenshot to the description showing the new configuration option in the UI. The default value is selected.

Figured this would be useful to @clement-fouque.

**Review by andrewkroh — APPROVED** [review]

**Comment by efd6** [comment]
Discussed with @clement-fouque offline.

**Pull request merged**